src/os/macosx/macos.h

```diff
@@ -40,4 +40,17 @@ bool IsMonospaceFont(CFStringRef name);
 
 void MacOSSetThreadName(const char *name);
 
+
+/** Deleter that calls CFRelease rather than deleting the pointer. */
+template <typename T> struct CFDeleter {
+	void operator()(T *p)
+	{
+		if (p) ::CFRelease(p);
+	}
+};
+
+/** Specialisation of std::unique_ptr for CoreFoundation objects. */
+template <typename T>
+using CFReleaser = std::unique_ptr<typename std::remove_pointer<T>::type, CFDeleter<typename std::remove_pointer<T>::type>>;
```

Comment:
Suggest naming this `CFAutoResource<>` and the functor doing the release call `CFDeleter<>`.

Reply:
I'd prefer something like `CFAutoRelease<>` more, but the naming is always some matter of opinion.

Except that, is the general idea sound?

Reply:
Oh, will it require a generic lambda? That needs C++17/20, iirc?